lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SseCustomerKey.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.s3;
+
+import static java.util.Objects.requireNonNull;
+import static software.amazon.awssdk.utils.BinaryUtils.fromBase64;
+import static software.amazon.awssdk.utils.Md5Utils.md5AsBase64;
+
+public record S3SseCustomerKey(String key, String md5, String algorithm)
+{
+    private static final String SSE_C_ALGORITHM = "AES256";
+
+    public S3SseCustomerKey
+    {
+        requireNonNull(key, "key is null");
+        requireNonNull(md5, "md5 is null");
+        requireNonNull(algorithm, "algorithm is null");
```

Comment:
will `algorithm` be always `AES256` for `S3SseCustomerKey`?

Reply:
although you can set different algorithm in sdk, the operation header (x-amz-server-side​-encryption​-customer-algorithm ) demands AES256 
https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerSideEncryptionCustomerKeys.html